src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -592,8 +593,17 @@ internal void UpdateCurrentSolutionOnOptionsChanged()
 #pragma warning disable IDE0060 // Remove unused parameter
     protected internal Task ScheduleTask(Action action, string? taskName = "Workspace.Task")
     {
-        _workQueue.AddWork(action);
-        return _workQueue.WaitUntilCurrentBatchCompletesAsync();
+        var handlerAndOptions = new WorkspaceEventHandlerAndOptions(args => action(), WorkspaceEventOptions.MainThreadDependent);
+        var handlerSet = EventHandlerSet.Create(handlerAndOptions);
+
+        return ScheduleTask(EventArgs.Empty, handlerSet);
+    }
+
+    [SuppressMessage("Style", "VSTHRD200:Use \"Async\" suffix for async methods", Justification = "This is a Task wrapper, not an asynchronous method.")]
+    internal Task ScheduleTask(EventArgs args, EventHandlerSet handlerSet)
```

Comment:
similar to before, do we need to expose the task outwards?

Reply:
still would like an answer on this.  no worries if we need this for now.  i just don't like it :)

Reply:
I didn't *want* to, but there are public API points that return this task (the other ScheduleTask methods, RaiseWorkspaceChangedEventAsync, RaiseTextDocumentOpenedOrClosedEventAsync, etc)